src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -94,7 +94,27 @@ bool is_output_buffer(const primitive_inst* prim, bool runtime_alloc) {
     }
     return false;
 }
+// TODO: Will remove macro when final confirm, remained it for the discussion
+#if 1
+bool is_user_cpu(const program_node* user) {
+    // If the user is dynamic and runtime skippable node, we still need its parents' completion event
+    // even though the user's program_node is can_be_optimized
+    if (user->can_be_optimized() && !(user->is_dynamic() && user->is_runtime_skippable()))
+        return false;
```

Comment:
`
    if (user->is_dynamic())
        return user->is_runtime_skippable();
    if (user->can_be_oiptimized()) {
        auto users = user->get_users();
        for (const auto& u : users) {
            if (is_user_cpu(u)) 
                return true;
        }
        bool is_cpu = user->get_selected_impl() ? user->get_selected_impl()->is_cpu() : user->Get_preferred_impl_type() == impl_types::cpu;
        return is_cpu;
    }
`

Reply:
Fixed it.